tests/api2/pool_dataset_encryption.py

```diff
@@ -0,0 +1,1006 @@
+        'encryption_options': {
+            'key': dataset_token_hex,
+        },
+        'encryption': True,
+        'inherit_encryption': False
+    }
+    results = POST('/pool/dataset/', payload)
+    assert results.status_code == 200, results.text
+    assert results.json()['key_format']['value'] == 'HEX', results.text
+
+
+@skip_for_ha
+def test_45_change_a_key_encrypted_root_to_passphrase_on_key_encrypted_pool():
+    payload = {
+        'id': dataset,
+        'change_key_options': {
+            'passphrase': 'my_passphrase',
+        }
+    }
+    results = POST(f'/pool/dataset/change_key/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+
+
+@skip_for_ha
+def test_46_verify_the_dataset_changed_to_passphrase():
+    results = GET(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+    assert results.json()['key_format']['value'] == 'PASSPHRASE', results.text
+
+
+@skip_for_ha
+def test_47_lock_passphrase_encrypted_dataset():
+    payload = {
+        'id': dataset,
+        'lock_options': {
+            'force_umount': True
+        }
+    }
+    results = POST('/pool/dataset/lock', payload)
+    assert results.status_code == 200, results.text
+
+
+@skip_for_ha
+def test_48_verify_the_dataset_is_locked():
+    results = GET(f'/pool/dataset/id/{dataset_url}/')
+    assert results.status_code == 200, results.text
+    assert results.json()['locked'] is True, results.text
+
+
+@skip_for_ha
+def test_49_verify_passphrase_encrypted_root_unlock_successful_is_false():
+    payload = {
+        'id': dataset
+    }
+    results = POST('/pool/dataset/encryption_summary/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
+    job_status_result = job_status['results']['result']
+    for dictionary in job_status_result:
+        if dictionary['name'] == dataset:
+            assert dictionary['unlock_successful'] is False, str(job_status_result)
+            assert dictionary['locked'] is True, str(job_status_result)
+            break
+    else:
+        assert False, str(job_status_result)
+
+
+@skip_for_ha
+def test_50_unlock_passphrase_encrypted_datasets():
+    payload = {
+        'id': dataset,
+        'unlock_options': {
+            'recursive': True,
+            'datasets': [
+                {
+                    'name': dataset,
+                    'passphrase': 'my_passphrase'
+                }
+            ]
+        }
+    }
+    results = POST('/pool/dataset/unlock/', payload)
+    assert results.status_code == 200, results.text
+    job_id = results.json()
+    job_status = wait_on_job(job_id, 120)
+    assert job_status['state'] == 'SUCCESS', str(job_status['results'])
```

Comment:
We should verify the call returns the datasets unlocked correctly ( like above )